packages/webapp/__tests__/SquadTokenPage.tsx

```diff
@@ -0,0 +1,294 @@
+import { OnboardingMode } from '@dailydotdev/shared/src/graphql/feed';
+import nock from 'nock';
+import AuthContext from '@dailydotdev/shared/src/contexts/AuthContext';
+import React from 'react';
+import { render, RenderResult, screen } from '@testing-library/preact';
+import { QueryClient, QueryClientProvider } from 'react-query';
+import { defaultTestSettings } from '@dailydotdev/shared/__tests__/fixture/settings';
+import { NextRouter } from 'next/router';
+import SettingsContext from '@dailydotdev/shared/src/contexts/SettingsContext';
+import { LoggedUser } from '@dailydotdev/shared/src/lib/user';
+import defaultUser from '@dailydotdev/shared/__tests__/fixture/loggedUser';
+import { Edge } from '@dailydotdev/shared/src/graphql/common';
+import {
+  MockedGraphQLResponse,
+  mockGraphQL,
+} from '@dailydotdev/shared/__tests__/helpers/graphql';
+import { waitForNock } from '@dailydotdev/shared/__tests__/helpers/utilities';
+import OnboardingContext from '@dailydotdev/shared/src/contexts/OnboardingContext';
+import { webappUrl } from '@dailydotdev/shared/src/lib/constants';
+import {
+  SquadInvitation,
+  SquadMember,
+  SquadMemberRole,
+  SQUAD_INVITATION_QUERY,
+  SQUAD_JOIN_MUTATION,
+} from '@dailydotdev/shared/src/graphql/squads';
+import SquadPage, {
+  SquadReferralProps,
+} from '../pages/squads/[handle]/[token]';
+
+const showLogin = jest.fn();
+let replaced = '';
+jest.mock('next/router', () => ({
+  useRouter: jest.fn().mockImplementation(
+    () =>
+      ({
+        isFallback: false,
+        // eslint-disable-next-line no-return-assign
+        replace: (path: string) => (replaced = path),
+      } as unknown as NextRouter),
+  ),
+}));
+
+beforeEach(() => {
+  jest.restoreAllMocks();
+  jest.clearAllMocks();
+  nock.cleanAll();
+  replaced = '';
+});
+
+let client: QueryClient;
+
+const defaultToken = 'ki3YLcxvSZ2Q6KgMBZvMbly1gnrZ6JnIrhTpUML-Hua';
+const generateMember = (i: string | number): SquadMember => ({
+  user: {
```

Comment:
Consider creating a fixture rather, it will help with future tests.

Reply:
I agree, though I think we can keep it a function, I had some issues earlier keeping it an object (modifying properties running across tests). Let me store this in a generic file.